rwengine/src/script/modules/GTA3ModuleImpl.inl

```diff
@@ -3878,31 +3878,53 @@ void opcode_0160(const ScriptArguments& args, ScriptVec3 coord, const ScriptChan
     @arg arg3 Boolean true/false
     @arg blip Blip
 */
-void opcode_0161(const ScriptArguments& args, const ScriptVehicle vehicle, const ScriptBlipColour arg2, const ScriptBoolean arg3, ScriptBlip& blip) {
-    RW_UNIMPLEMENTED_OPCODE(0x0161);
-    RW_UNUSED(vehicle);
-    RW_UNUSED(arg2);
-    RW_UNUSED(arg3);
-    RW_UNUSED(blip);
-    RW_UNUSED(args);
+void opcode_0161(const ScriptArguments& args, const ScriptVehicle vehicle, const ScriptBlipColour colour, const ScriptBlipDisplay display, ScriptBlip& blip) {
+    auto& data = script::createObjectBlip(args, vehicle);
+    data.colour = colour;
+    /// @todo change ScriptBlipDisplay to BlipData::DisplayMode
+    data.display = static_cast <BlipData::DisplayMode> (display);
+    blip = &data;
 }
 
 /**
-    @brief %4d% = create_marker_above_actor %1d% color %2d% visibility %3d%
+    @brief ADD_BLIP_FOR_CHAR_OLD pedHandle colour display outblipHandle
 
-    opcode 0162
-    @arg character Character/ped
-    @arg arg2 
-    @arg arg3 
-    @arg blip Blip
+    opcode 0162 / 354
+
+    Creates and attaches blip for an ped
+
+    @arg pedHandle Ped
+    @arg colour Blip colour
+    @arg display Blip display mode
+    @arg outblipHandle Created blip
 */
-void opcode_0162(const ScriptArguments& args, const ScriptCharacter character, const ScriptBlipColour arg2, const ScriptBlipDisplay arg3, ScriptBlip& blip) {
-    RW_UNIMPLEMENTED_OPCODE(0x0162);
-    RW_UNUSED(character);
-    RW_UNUSED(arg2);
-    RW_UNUSED(arg3);
-    RW_UNUSED(blip);
-    RW_UNUSED(args);
+void opcode_0162(const ScriptArguments& args, const ScriptCharacter character, const ScriptBlipColour colour, const ScriptBlipDisplay display, ScriptBlip& blip) {
+    auto& data = script::createObjectBlip(args, character);
+    data.colour = colour;
+    /// @todo change ScriptBlipDisplay to BlipData::DisplayMode
+    data.display = static_cast <BlipData::DisplayMode> (display);
+    blip = &data;
+}
+
+/**
+    @brief ADD_BLIP_FOR_OBJECT_OLD objectHandle colour display outblipHandle
+
+    opcode 0163 / 355
```

Comment:
For this comment block, see other instance above.
Also take care of other instances which follow.